subprojects/docs/src/docs/release/notes.md

```diff
@@ -90,6 +90,11 @@ Changes to the PATH will be visible to `Exec` tasks and calling `System.getenv`
 
 However, we strongly recommend that build and plugin authors use Gradle properties instead of `System.getenv` for a more idiomatic end user experience.
 
+### Incremental build uses less memory
+
+Some improvements made to up-to-date checking results in less heap space being used.
+For a build similar to Gradle's own this results in a 60 MB reduction in heap usage.
```

Comment:
It would be nice to know how much heap the Gradle build retains between invocations to put that number into perspective. WDYT?

Reply:
~450MB after all the improvements, so the 60MB were a ~13% reduction